File: movie_backend/frontend/src/services/collectionService.js
import axios from "axios";

const BASE_URL = "http://127.0.0.1:8000";

// =========================
// GET collections by user
// =========================
export const getCollections = (user_id) => {
  return axios.get(`${BASE_URL}/api/collections/user/${user_id}`);
};

// =========================
// CREATE collection
// =========================
export const createCollection = (user_id, data) => {
  return axios.post(
    `${BASE_URL}/api/collections/?user_id=${user_id}`,
    data
  );
};

// =========================
// DELETE collection
// =========================
export const deleteCollection = (collection_id) => {
  return axios.delete(
    `${BASE_URL}/api/collections/${collection_id}`
  );
};

// =========================
// ADD movie
// =========================
export const addMovieToCollection = (collection_id, data) => {
  return axios.post(
    `${BASE_URL}/api/collections/${collection_id}/movies`,
    data
  );
};

// =========================
// REMOVE movie
// =========================
export const removeMovieFromCollection = (collection_id, movie_id) => {
  return axios.delete(
    `${BASE_URL}/api/collections/${collection_id}/movies/${movie_id}`
  );
};

File: movie_backend/app/services/collection_service.py
from uuid import uuid4

collections_db = {}

def create_collection(user_id, data):
    collection_id = str(uuid4())

    new_collection = {
        "collection_id": collection_id,
        "user_id": user_id,
        "name": data.name,
        "description": data.description,
        "movies": []
    }

    collections_db[collection_id] = new_collection
    return new_collection


def get_collections(user_id):
    return [
        c for c in collections_db.values()
        if c["user_id"] == user_id
    ]


def delete_collection(collection_id):
    return collections_db.pop(collection_id, None)


def add_movie(collection_id, movie):
    if collection_id in collections_db:
        collections_db[collection_id]["movies"].append(movie)
        return collections_db[collection_id]
    return None